### PR TITLE
Update Rust nightly version

### PR DESCRIPTION
**Pull request by bradjc**
### Pull Request Overview

It has been about 3 months, so a good time to stay current. Also motivated by https://github.com/rust-lang/rust/pull/67887 which should make debugging easier.

Changes
- New warning about `()` not needed.
- Update to update script.

### Testing Strategy

travis

### TODO or Help Wanted

n/a

### Documentation Updated

- [x] Updated the relevant files in `/docs`, or no updates are required.

### Formatting

- [x] Ran `make formatall`.

### CODE REVIEW

File: boards/Makefile.common [hunk 1]
```diff
@@ -148,7 +148,7 @@ debug-lst: target/$(TARGET)/debug/$(PLATFORM).lst
 
 .PHONY: doc
 doc: | target
-	$(Q)RUSTDOCFLAGS=--document-private-items $(CARGO) doc $(VERBOSE) --release --target=$(TARGET)
```
Comment (gendx) [inline boards/Makefile.common]: I'm not sure what was the meaning of this flag, but could you explained why you had to remove it?

Comment (gendx) [reply boards/Makefile.common]: Ok, just found the explanation in the commit history.

**Comment by gendx** [comment]
> Regarding [rust-lang/rust#67887](https://github.com/rust-lang/rust/pull/67887) (which sounds quite useful for debugging by the way), could you document what is the increase in code size for the various boards?

I had a quick look, and it seems that overall code and/or data size decreased on all boards with this change :)

**Comment by hudson-ayers** [comment]
> > Regarding [rust-lang/rust#67887](https://github.com/rust-lang/rust/pull/67887) (which sounds quite useful for debugging by the way), could you document what is the increase in code size for the various boards?
> 
> I had a quick look, and it seems that overall code and/or data size decreased on all boards with this change :)

I think that the primary reason that we currently use `.expect()` rather than `.unwrap()` was to make it easy to trace the source of panics, so with this change I would be all for changing most `.expect()` calls back to `.unwrap()` which should reduce data size by removing storage for the strings passed to `.except()`

File: tools/update_rust_version.sh [hunk 2]
```diff
@@ -4,13 +4,21 @@
...
+# RUSTUP_NIGHTLY_VERSION=`rustup update nightly 2>&1`
+# BEST_DATE=`echo $RUSTUP_NIGHTLY_VERSION | sed 's/.* \([0-9]*-[0-9]*-[0-9]*\).*/\1/g'`
+
+# I just do not know how to get rustup to tell us the version of the toolchain
+# it decided on with the format required for `rust-toolchain`. That the dates
+# are off-by-one day is annoying. I'm resorting to just asking the user.
+
+echo "Please enter the version of Rust to use."
+echo "It is probably just one day later than whatever was printed out above."
+read BEST_DATE
```
Comment (ppannuto) [inline tools/update_rust_version.sh]: This is unfortunate, but really probably only run by Tock core folks, so not really a problem

**Pull request merged**